share/man/man1/brew.1
.\" generated with Ronn/v0.7.3
.\" http://github.com/rtomayko/ronn/tree/0.7.3
.
.TH "BREW" "1" "September 2015" "Tigerbrew" "brew"
.
.SH "NAME"
\fBbrew\fR \- The missing package manager for OS X
.
.SH "SYNOPSIS"
\fBbrew\fR \-\-version
.
.br
\fBbrew\fR command [\-\-verbose|\-v] [options] [formula] \.\.\.
.
.SH "DESCRIPTION"
Tigerbrew is the easiest and most flexible way to install the UNIX tools Apple didn\'t include with OS X\.
.
.SH "ESSENTIAL COMMANDS"
For the full command list, see the COMMANDS section\.
.
.P
With \fB\-\-verbose\fR or \fB\-v\fR, many commands print extra debugging information\. Note that these flags should only appear after a command\.
.
.TP
\fBinstall\fR \fIformula\fR
Install \fIformula\fR\.
.
.TP
\fBremove\fR \fIformula\fR
Uninstall \fIformula\fR\.
.
.TP
\fBupdate\fR
Fetch the newest version of Tigerbrew from GitHub using \fBgit\fR(1)\.
.
.TP
\fBlist\fR
List all installed formulae\.
.
.TP
\fBsearch\fR \fItext\fR|/\fItext\fR/
Perform a substring search of formula names for \fItext\fR\. If \fItext\fR is surrounded with slashes, then it is interpreted as a regular expression\. The search for \fItext\fR is extended online to some popular taps\. If no search term is given, all locally available formulae are listed\.
.
.SH "COMMANDS"
.
.IP "\(bu" 4
\fBaudit\fR [\-\-strict] [\-\-online] [\fIformulae\fR]: Check \fIformulae\fR for Tigerbrew coding style violations\. This should be run before submitting a new formula\.
.
.IP
If no \fIformulae\fR are provided, all of them are checked\.
.
.IP
If \fB\-\-strict\fR is passed, additional checks are run\. This should be used when creating for new formulae\.
.
.IP
If \fB\-\-online\fR is passed, additional slower checks that require a network connection are run\. This should be used when creating for new formulae\.
.
.IP
\fBaudit\fR exits with a non\-zero status if any errors are found\. This is useful, for instance, for implementing pre\-commit hooks\.
.
.IP "\(bu" 4
\fBcat\fR \fIformula\fR: Display the source to \fIformula\fR\.
.
.IP "\(bu" 4
\fBcleanup [\-\-force] [\-\-prune=<days>] [\-ns]\fR [\fIformulae\fR]: For all installed or specific formulae, remove any older versions from the cellar\. By default, does not remove out\-of\-date keg\-only brews, as other software may link directly to specific versions\. In addition old downloads from the Tigerbrew download\-cache are deleted\.
.
.IP
If \fB\-\-force\fR is passed, remove out\-of\-date keg\-only brews as well\.
.
.IP
If \fB\-\-prune=<days>\fR is specified, remove all cache files older than \fIdays\fR\.
.
.IP
If \fB\-n\fR is passed, show what would be removed, but do not actually remove anything\.
.
.IP
If \fB\-s\fR is passed, scrubs the cache, removing downloads for even the latest versions of formula\. Note downloads for any installed formula will still not be deleted\. If you want to delete those too: \fBrm \-rf $(brew \-\-cache)\fR
.
.IP "\(bu" 4
\fBcommand\fR \fIcmd\fR: Display the path to the file which is used when invoking \fBbrew <cmd>\fR\.
.
.IP "\(bu" 4
\fBcommands [\-\-quiet [\-\-include\-aliases]]\fR: Show a list of built\-in and external commands\.
.
.IP
If \fB\-\-quiet\fR is passed, list only the names of commands without the header\. With \fB\-\-include\-aliases\fR, the aliases of internal commands will be included\.
.
.IP "\(bu" 4
\fBconfig\fR: Show Tigerbrew and system configuration useful for debugging\. If you file a bug report, you will likely be asked for this information if you do not provide it\.
.
.IP "\(bu" 4
\fBcreate <URL> [\-\-autotools|\-\-cmake] [\-\-no\-fetch] [\-\-set\-name <name>] [\-\-set\-version <version>]\fR: Generate a formula for the downloadable file at \fIURL\fR and open it in the editor\. Tigerbrew will attempt to automatically derive the formula name and version, but if it fails, you\'ll have to make your own template\. The wget formula serves as a simple example\. For a complete cheat\-sheet, have a look at
.
.IP
\fIhttp://www\.rubydoc\.info/github/Homebrew/homebrew/master/frames\fR
.
.IP
If \fB\-\-autotools\fR is passed, create a basic template for an Autotools\-style build\. If \fB\-\-cmake\fR is passed, create a basic template for a CMake\-style build\.
.
.IP
If \fB\-\-no\-fetch\fR is passed, Tigerbrew will not download \fIURL\fR to the cache and will thus not add the SHA256 to the formula for you\.
.
.IP
The options \fB\-\-set\-name\fR and \fB\-\-set\-version\fR each take an argument and allow you to explicitly set the name and version of the package you are creating\.
.
.IP "\(bu" 4
\fBdeps [\-\-1] [\-n] [\-\-union] [\-\-tree] [\-\-all] [\-\-installed] [\-\-skip\-build] [\-\-skip\-optional]\fR \fIformulae\fR: Show dependencies for \fIformulae\fR\. When given multiple formula arguments, show the intersection of dependencies for \fIformulae\fR, except when passed \fB\-\-tree\fR, \fB\-\-all\fR, or \fB\-\-installed\fR\.
.
.IP
If \fB\-\-1\fR is passed, only show dependencies one level down, instead of recursing\.
.
.IP
If \fB\-n\fR is passed, show dependencies in topological order\.
.
.IP
If \fB\-\-union\fR is passed, show the union of dependencies for \fIformulae\fR, instead of the intersection\.
.
.IP
If \fB\-\-tree\fR is passed, show dependencies as a tree\.
.
.IP
If \fB\-\-all\fR is passed, show dependencies for all formulae\.
.
.IP
If \fB\-\-installed\fR is passed, show dependencies for all installed formulae\.
.
.IP
By default, \fBdeps\fR shows dependencies for \fIformulae\fR\. To skip the \fB:build\fR type dependencies, pass \fB\-\-skip\-build\fR\. Similarly, pass \fB\-\-skip\-optional\fR to skip \fB:optional\fR dependencies\.
.
.IP "\(bu" 4
\fBdesc\fR \fIformula\fR: Display \fIformula\fR\'s name and one\-line description\.
.
.IP "\(bu" 4
\fBdesc [\-s|\-n|\-d] <pattern>\fR: Search both name and description (\fB\-s\fR), just the names (\fB\-n\fR), or just the descriptions (\fB\-d\fR) for \fB<pattern>\fR\. \fB<pattern>\fR is by default interpreted as a literal string; if flanked by slashes, it is instead interpreted as a regular expression\. Formula descriptions are cached; the cache is created on the first search, making that search slower than subsequent ones\.
.
.IP "\(bu" 4
\fBdiy [\-\-name=<name>] [\-\-version=<version>]\fR: Automatically determine the installation prefix for non\-Tigerbrew software\.
.
.IP
Using the output from this command, you can install your own software into the Cellar and then link it into Tigerbrew\'s prefix with \fBbrew link\fR\.
.
.IP
The options \fB\-\-name=<name>\fR and \fB\-\-version=<version>\fR each take an argument and allow you to explicitly set the name and version of the package you are installing\.
.
.IP "\(bu" 4
\fBdoctor\fR: Check your system for potential problems\. Doctor exits with a non\-zero status if any problems are found\.
.
.IP "\(bu" 4
\fBedit\fR: Open all of Tigerbrew for editing\.
.
.IP "\(bu" 4
\fBedit\fR \fIformula\fR: Open \fIformula\fR in the editor\.
.
.IP "\(bu" 4
\fBfetch [\-\-force] [\-v] [\-\-devel|\-\-HEAD] [\-\-deps] [\-\-build\-from\-source|\-\-force\-bottle]\fR \fIformulae\fR: Download the source packages for the given \fIformulae\fR\. For tarballs, also print SHA1 and SHA\-256 checksums\.
.
.IP
If \fB\-\-HEAD\fR or \fB\-\-devel\fR is passed, fetch that version instead of the stable version\.
.
.IP
If \fB\-v\fR is passed, do a verbose VCS checkout, if the URL represents a CVS\. This is useful for seeing if an existing VCS cache has been updated\.
.
.IP
If \fB\-\-force\fR is passed, remove a previously cached version and re\-fetch\.
.
.IP
If \fB\-\-deps\fR is passed, also download dependencies for any listed \fIformulae\fR\.
.
.IP
If \fB\-\-build\-from\-source\fR is passed, download the source rather than a bottle\.
.
.IP
If \fB\-\-force\-bottle\fR is passed, download a bottle if it exists for the current version of OS X, even if it would not be used during installation\.
.
.IP "\(bu" 4
\fBhome\fR: Open Tigerbrew\'s own homepage in a browser\.
.
.IP "\(bu" 4
\fBhome\fR \fIformula\fR: Open \fIformula\fR\'s homepage in a browser\.
.
.IP "\(bu" 4
\fBinfo\fR \fIformula\fR: Display information about \fIformula\fR\.
.
.IP "\(bu" 4
\fBinfo \-\-github\fR \fIformula\fR: Open a browser to the GitHub History page for formula \fIformula\fR\.
.
.IP
To view formula history locally: \fBbrew log \-p <formula>\fR\.
.
.IP "\(bu" 4
\fBinfo \-\-json=<version>\fR (\-\-all|\-\-installed|\fIformulae\fR): Print a JSON representation of \fIformulae\fR\. Currently the only accepted value for \fIversion\fR is \fBv1\fR\.
.
.IP
Pass \fB\-\-all\fR to get information on all formulae, or \fB\-\-installed\fR to get information on all installed formulae\.
.
.IP
See the docs for examples of using the JSON: \fIhttps://github\.com/mistydemeo/tigerbrew/blob/master/share/doc/homebrew/Querying\-Brew\.md\fR
.
.IP "\(bu" 4
\fBinstall [\-\-debug] [\-\-env=<std|super>] [\-\-ignore\-dependencies] [\-\-only\-dependencies] [\-\-cc=<compiler>] [\-\-build\-from\-source|\-\-force\-bottle] [\-\-devel|\-\-HEAD]\fR \fIformula\fR: Install \fIformula\fR\.
.
.IP
\fIformula\fR is usually the name of the formula to install, but it can be specified several different ways\. See \fISPECIFYING FORMULAE\fR\.
.
.IP
If \fB\-\-debug\fR is passed and brewing fails, open an interactive debugging session with access to IRB or a shell inside the temporary build directory\.
.
.IP
If \fB\-\-env=std\fR is passed, use the standard build environment instead of superenv\.
.
.IP
If \fB\-\-env=super\fR is passed, use superenv even if the formula specifies the standard build environment\.
.
.IP
If \fB\-\-ignore\-dependencies\fR is passed, skip installing any dependencies of any kind\. If they are not already present, the formula will probably fail to install\.
.
.IP
If \fB\-\-only\-dependencies\fR is passed, install the dependencies with specified options but do not install the specified formula\.
.
.IP
If \fB\-\-cc=<compiler>\fR is passed, attempt to compile using \fIcompiler\fR\. \fIcompiler\fR should be the name of the compiler\'s executable, for instance \fBgcc\-4\.2\fR for Apple\'s GCC 4\.2, or \fBgcc\-4\.9\fR for a Tigerbrew\-provided GCC 4\.9\.
.
.IP
If \fB\-\-build\-from\-source\fR is passed, compile from source even if a bottle is provided for \fIformula\fR\.
.
.IP
If \fB\-\-force\-bottle\fR is passed, install from a bottle if it exists for the current version of OS X, even if custom options are given\.
.
.IP
If \fB\-\-devel\fR is passed, and \fIformula\fR defines it, install the development version\.
.
.IP
If \fB\-\-HEAD\fR is passed, and \fIformula\fR defines it, install the HEAD version, aka master, trunk, unstable\.
.
.IP
To install a newer version of HEAD use \fBbrew rm <foo> && brew install \-\-HEAD <foo>\fR\.
.
.IP "\(bu" 4
\fBinstall \-\-interactive [\-\-git]\fR \fIformula\fR: Download and patch \fIformula\fR, then open a shell\. This allows the user to run \fB\./configure \-\-help\fR and otherwise determine how to turn the software package into a Tigerbrew formula\.
.
.IP
If \fB\-\-git\fR is passed, Tigerbrew will create a Git repository, useful for creating patches to the software\.
.
.IP "\(bu" 4
\fBirb [\-\-examples]\fR: Enter the interactive Tigerbrew Ruby shell\.
.
.IP
If \fB\-\-examples\fR is passed, several examples will be shown\.
.
.IP "\(bu" 4
\fBleaves\fR: Show installed formulae that are not dependencies of another installed formula\.
.
.IP "\(bu" 4
\fBln\fR, \fBlink [\-\-overwrite] [\-\-dry\-run] [\-\-force]\fR \fIformula\fR: Symlink all of \fIformula\fR\'s installed files into the Tigerbrew prefix\. This is done automatically when you install formulae but can be useful for DIY installations\.
.
.IP
If \fB\-\-overwrite\fR is passed, Tigerbrew will delete files which already exist in the prefix while linking\.
.
.IP
If \fB\-\-dry\-run\fR or \fB\-n\fR is passed, Tigerbrew will list all files which would be linked or which would be deleted by \fBbrew link \-\-overwrite\fR, but will not actually link or delete any files\.
.
.IP
If \fB\-\-force\fR is passed, Tigerbrew will allow keg\-only formulae to be linked\.
.
.IP "\(bu" 4
\fBlinkapps [\-\-local]\fR [\fIformulae\fR]: Find installed formulae that have compiled \fB\.app\fR\-style "application" packages for OS X, and symlink those apps into \fB/Applications\fR, allowing for easier access\.
.
.IP
If no \fIformulae\fR are provided, all of them will have their \.apps symlinked\.
.
.IP
If provided, \fB\-\-local\fR will move them into the user\'s \fB~/Applications\fR directory instead of the system directory\. It may need to be created, first\.
.
.IP "\(bu" 4
\fBls\fR, \fBlist [\-\-full\-name]\fR List all installed formulae\. If \fB\-\-full\-name\fR is passed, print formulae with full\-qualified names\.
.
.IP "\(bu" 4
\fBls\fR, \fBlist \-\-unbrewed\fR List all files in the Tigerbrew prefix not installed by Tigerbrew\.
.
.IP "\(bu" 4
\fBls\fR, \fBlist [\-\-versions [\-\-multiple]] [\-\-pinned]\fR [\fIformulae\fR]: List the installed files for \fIformulae\fR\. Combined with \fB\-\-verbose\fR, recursively list the contents of all subdirectories in each \fIformula\fR\'s keg\.
.
.IP
If \fB\-\-versions\fR is passed, show the version number for installed formulae, or only the specified formulae if \fIformulae\fR are given\. With \fB\-\-multiple\fR, only show formulae with multiple versions installed\.
.
.IP
If \fB\-\-pinned\fR is passed, show the versions of pinned formulae, or only the specified (pinned) formulae if \fIformulae\fR are given\. See also \fBpin\fR, \fBunpin\fR\.
.
.IP "\(bu" 4
\fBlog [git\-log\-options]\fR \fIformula\fR \.\.\.: Show the git log for the given formulae\. Options that \fBgit\-log\fR(1) recognizes can be passed before the formula list\.
.
.IP "\(bu" 4
\fBmissing\fR [\fIformulae\fR]: Check the given \fIformulae\fR for missing dependencies\.
.
.IP
If no \fIformulae\fR are given, check all installed brews\.
.
.IP "\(bu" 4
\fBmigrate [\-\-force]\fR \fIformulae\fR: Migrate renamed packages to new name, where \fIformulae\fR are old names of packages\.
.
.IP
If \fB\-\-force\fR is passed, then treat installed \fIformulae\fR and passed \fIformulae\fR like if they are from same taps and migrate them anyway\.
.
.IP "\(bu" 4
\fBoptions [\-\-compact] [\-\-all] [\-\-installed]\fR \fIformula\fR: Display install options specific to \fIformula\fR\.
.
.IP
If \fB\-\-compact\fR is passed, show all options on a single line separated by spaces\.
.
.IP
If \fB\-\-all\fR is passed, show options for all formulae\.
.
.IP
If \fB\-\-installed\fR is passed, show options for all installed formulae\.
.
.IP "\(bu" 4
\fBoutdated [\-\-quiet | \-\-verbose | \-\-json=v1 ]\fR: Show formulae that have an updated version available\.
.
.IP
By default, version information is displayed in interactive shells, and suppressed otherwise\.
.
.IP
If \fB\-\-quiet\fR is passed, list only the names of outdated brews (takes precedence over \fB\-\-verbose\fR)\.
.
.IP
If \fB\-\-verbose\fR is passed, display detailed version information\.
.
.IP
If \fB\-\-json=<version>\fR is passed, the output will be in JSON format\. The only valid version is \fBv1\fR\.
.
.IP "\(bu" 4
\fBpin\fR \fIformulae\fR: Pin the specified \fIformulae\fR, preventing them from being upgraded when issuing the \fBbrew upgrade\fR command\. See also \fBunpin\fR\.
.
.IP "\(bu" 4
\fBprune\fR: Remove dead symlinks from the Tigerbrew prefix\. This is generally not needed, but can be useful when doing DIY installations\.
.
.IP "\(bu" 4
\fBreinstall\fR \fIformula\fR: Uninstall then install \fIformula\fR
.
.IP "\(bu" 4
\fBrm\fR, \fBremove\fR, \fBuninstall [\-\-force]\fR \fIformula\fR: Uninstall \fIformula\fR\.
.
.IP
If \fB\-\-force\fR is passed, and there are multiple versions of \fIformula\fR installed, delete all installed versions\.
.
.IP "\(bu" 4
\fBsearch\fR, \fB\-S\fR: Display all locally available formulae for brewing (including tapped ones)\. No online search is performed if called without arguments\.
.
.IP "\(bu" 4
\fBsearch\fR, \fB\-S\fR \fItext\fR|/\fItext\fR/: Perform a substring search of formula names for \fItext\fR\. If \fItext\fR is surrounded with slashes, then it is interpreted as a regular expression\. The search for \fItext\fR is extended online to some popular taps\.
.
.IP "\(bu" 4
\fBsearch \-\-debian\fR|\fB\-\-fedora\fR|\fB\-\-fink\fR|\fB\-\-macports\fR|\fB\-\-opensuse\fR|\fB\-\-ubuntu\fR \fItext\fR: Search for \fItext\fR in the given package manager\'s list\.
.
.IP "\(bu" 4
\fBsh [\-\-env=std]\fR: Instantiate a Tigerbrew build environment\. Uses our years\-battle\-hardened Tigerbrew build logic to help your \fB\./configure && make && make install\fR or even your \fBgem install\fR succeed\. Especially handy if you run Tigerbrew in a Xcode\-only configuration since it adds tools like make to your PATH which otherwise build\-systems would not find\.
.
.IP "\(bu" 4
\fBswitch\fR \fIname\fR \fIversion\fR: Symlink all of the specific \fIversion\fR of \fIname\fR\'s install to Tigerbrew prefix\.
.
.IP "\(bu" 4
\fBtap\fR [\-\-full] [<user/repo>] [\fIURL\fR]: Tap a formula repository or list existing taps\. This command can be invoked in three ways\.
.
.IP "\(bu" 4
\fBtap\fR without arguments displays existing taps\.
.
.IP "\(bu" 4
\fBtap <user/repo>\fR taps a formula repository from GitHub using HTTPS\. Since so many taps are hosted on GitHub, this command is a shortcut for \fBtap user/repo https://github\.com/#{user}/homebrew\-#{repo}\fR\.
.
.IP "\(bu" 4
\fBtap <user/repo> <URL>\fR taps a formula repository from anywhere, using any transport protocol that \fBgit\fR handles\. The one\-argument form of \fBtap\fR simplifies but also limits\. This two\-argument command makes no assumptions, so taps can be cloned from places other than GitHub and using protocols other than HTTPS, e\.g\., SSH, GIT, HTTP, FTP(S), RSYNC\.
.
.IP "" 0
.
.IP
By default, the repository is cloned as a shallow copy (\fB\-\-depth=1\fR), but if \fB\-\-full\fR is passed, a full clone will be used\.
.
.IP "\(bu" 4
\fBtap \-\-repair\fR: Migrate tapped formulae from symlink\-based to directory\-based structure\.
.
.IP "\(bu" 4
\fBtap \-\-list\-official\fR: List all official taps\.
.
.IP "\(bu" 4
\fBtap \-\-list\-pinned\fR: List all pinned taps\.
.
.IP "\(bu" 4
\fBtap\-info\fR \fItap\fR: Display information about \fItap\fR\.
.
.IP "\(bu" 4
\fBtap\-info \-\-json=<version>\fR (\-\-installed|\fItaps\fR): Print a JSON representation of \fItaps\fR\. Currently the only accepted value for \fIversion\fR is \fBv1\fR\.
.
.IP
Pass \fB\-\-installed\fR to get information on installed taps\.
.
.IP
See the docs for examples of using the JSON: \fIhttps://github\.com/mistydemeo/tigerbrew/blob/master/share/doc/homebrew/Querying\-Brew\.md\fR
.
.IP "\(bu" 4
\fBtap\-pin\fR \fItap\fR: Pin \fItap\fR, prioritizing its formulae over core when formula names are supplied by the user\. See also \fBtap\-unpin\fR\.
.
.IP "\(bu" 4
\fBtap\-unpin\fR \fItap\fR: Unpin \fItap\fR so its formulae are no longer prioritized\. See also \fBtap\-pin\fR\.
.
.IP "\(bu" 4
\fBtest\fR [\-\-devel|\-\-HEAD] [\-\-debug] \fIformula\fR: A few formulae provide a test method\. \fBbrew test <formula>\fR runs this test method\. There is no standard output or return code, but it should generally indicate to the user if something is wrong with the installed formula\.
.
.IP
To test the development or head version of a formula, use \fB\-\-devel\fR or \fB\-\-HEAD\fR\.
.
.IP
If \fB\-\-debug\fR is passed and the test fails, an interactive debugger will be launched with access to IRB or a shell inside the temporary test directory\.
.
.IP
Example: \fBbrew install jruby && brew test jruby\fR
.
.IP "\(bu" 4
\fBunlink [\-\-dry\-run]\fR \fIformula\fR: Remove symlinks for \fIformula\fR from the Tigerbrew prefix\. This can be useful for temporarily disabling a formula: \fBbrew unlink foo && commands && brew link foo\fR\.
.
.IP
If \fB\-\-dry\-run\fR or \fB\-n\fR is passed, Homebrew will list all files which would be unlinked, but will not actually unlink or delete any files\.
.
.IP "\(bu" 4
\fBunlinkapps [\-\-local]\fR [\fIformulae\fR]: Removes links created by \fBbrew linkapps\fR\.
.
.IP
If no \fIformulae\fR are provided, all linked app will be removed\.
.
.IP "\(bu" 4
\fBunpack [\-\-git|\-\-patch] [\-\-destdir=<path>]\fR \fIformulae\fR: Unpack the source files for \fIformulae\fR into subdirectories of the current working directory\. If \fB\-\-destdir=<path>\fR is given, the subdirectories will be created in the directory named by \fB<path>\fR instead\.
.
.IP
If \fB\-\-patch\fR is passed, patches for \fIformulae\fR will be applied to the unpacked source\.
.
.IP
If \fB\-\-git\fR is passed, a Git repository will be initalized in the unpacked source\. This is useful for creating patches for the software\.
.
.IP "\(bu" 4
\fBunpin\fR \fIformulae\fR: Unpin \fIformulae\fR, allowing them to be upgraded by \fBbrew upgrade\fR\. See also \fBpin\fR\.
.
.IP "\(bu" 4
\fBuntap\fR \fItap\fR: Remove a tapped repository\.
.
.IP "\(bu" 4
\fBupdate [\-\-rebase]\fR: Fetch the newest version of Tigerbrew and all formulae from GitHub using \fBgit\fR(1)\.
.
.IP
If \fB\-\-rebase\fR is specified then \fBgit pull \-\-rebase\fR is used\.
.
.IP "\(bu" 4
\fBupgrade [install\-options]\fR [\fIformulae\fR]: Upgrade outdated, unpinned brews\.
.
.IP
Options for the \fBinstall\fR command are also valid here\.
.
.IP
If \fIformulae\fR are given, upgrade only the specified brews (but do so even if they are pinned; see \fBpin\fR, \fBunpin\fR)\.
.
.IP "\(bu" 4
\fBuses [\-\-installed] [\-\-recursive] [\-\-skip\-build] [\-\-skip\-optional] [\-\-devel|\-\-HEAD]\fR \fIformulae\fR: Show the formulae that specify \fIformulae\fR as a dependency\. When given multiple formula arguments, show the intersection of formulae that use \fIformulae\fR\.
.
.IP
Use \fB\-\-recursive\fR to resolve more than one level of dependencies\.
.
.IP
If \fB\-\-installed\fR is passed, only list installed formulae\.
.
.IP
By default, \fBuses\fR shows all formulae that specify \fIformulae\fR as a dependency\. To skip the \fB:build\fR type dependencies, pass \fB\-\-skip\-build\fR\. Similarly, pass \fB\-\-skip\-optional\fR to skip \fB:optional\fR dependencies\.
.
.IP
By default, \fBuses\fR shows usages of \fBformula\fR by stable builds\. To find cases where \fBformula\fR is used by development or HEAD build, pass \fB\-\-devel\fR or \fB\-\-HEAD\fR\.
.
.IP "\(bu" 4
\fB\-\-cache\fR: Display Tigerbrew\'s download cache\. See also \fBHOMEBREW_CACHE\fR\.
.
.IP "\(bu" 4
\fB\-\-cache\fR \fIformula\fR: Display the file or directory used to cache \fIformula\fR\.
.
.IP "\(bu" 4
\fB\-\-cellar\fR: Display Tigerbrews\'s Cellar path\. \fIDefault:\fR \fB$(brew \-\-prefix)/Cellar\fR, or if that directory doesn\'t exist, \fB$(brew \-\-repository)/Cellar\fR\.
.
.IP "\(bu" 4
\fB\-\-cellar\fR \fIformula\fR: Display the location in the cellar where \fIformula\fR would be installed, without any sort of versioned directory as the last path\.
.
.IP "\(bu" 4
\fB\-\-env\fR: Show a summary of the Tigerbrews build environment\.
.
.IP "\(bu" 4
\fB\-\-prefix\fR: Display Tigerbrews\'s install path\. \fIDefault:\fR \fB/usr/local\fR
.
.IP "\(bu" 4
\fB\-\-prefix\fR \fIformula\fR: Display the location in the cellar where \fIformula\fR is or would be installed\.
.
.IP "\(bu" 4
\fB\-\-repository\fR: Display where Tigerbrews\'s \fB\.git\fR directory is located\. For standard installs, the \fBprefix\fR and \fBrepository\fR are the same directory\.
.
.IP "\(bu" 4
\fB\-\-version\fR: Print the version number of brew to standard error and exit\.
.
.IP "" 0
.
.SH "EXTERNAL COMMANDS"
Tigerbrews, like \fBgit\fR(1), supports external commands\. These are executable scripts that reside somewhere in the PATH, named \fBbrew\-<cmdname>\fR or \fBbrew\-<cmdname>\.rb\fR, which can be invoked like \fBbrew cmdname\fR\. This allows you to create your own commands without modifying Tigerbrews\'s internals\.
.
.P
Instructions for creating your own commands can be found in the docs: \fIhttps://github\.com/mistydemeo/tigerbrew/blob/master/share/doc/homebrew/External\-Commands\.md\fR
.
.SH "SPECIFYING FORMULAE"
Many Tigerbrew commands accept one or more \fIformula\fR arguments\. These arguments can take several different forms:
.
.TP
The name of a formula
e\.g\. \fBgit\fR, \fBnode\fR, \fBwget\fR\.
.
.TP
The fully\-qualified name of a tapped formula
Sometimes a formula from a tapped repository may conflict with one in mistydemeo/tigerbrew\. You can still access these formulae by using a special syntax, e\.g\. \fBhomebrew/dupes/vim\fR or \fBhomebrew/versions/node4\fR\.
.
.TP
An arbitrary URL
Tigerbrew can install formulae via URL, e\.g\. \fBhttps://raw\.github\.com/mistydemeo/tigerbrew/master/Library/Formula/git\.rb\fR\. The formula file will be cached for later use\.
.
.SH "ENVIRONMENT"
.
.TP
AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY
When using the S3 download strategy, Tigerbrew will look in these variables for access credentials (see \fIhttps://docs\.aws\.amazon\.com/cli/latest/userguide/cli\-chap\-getting\-started\.html#cli\-environment\fR to retrieve these access credentials from AWS)\. If they are not set, the S3 download strategy will download with a public (unsigned) URL\.
.
.TP
BROWSER
If set, and \fBHOMEBREW_BROWSER\fR is not, use \fBBROWSER\fR as the web browser when opening project homepages\.
.
.TP
EDITOR

Library/brew.rb
#!/usr/bin/ruby -W0
# -*- coding: utf-8 -*-

std_trap = trap("INT") { exit! 130 } # no backtrace thanks

HOMEBREW_BREW_FILE = ENV["HOMEBREW_BREW_FILE"]

if ARGV == %w[--prefix]
  puts File.dirname(File.dirname(HOMEBREW_BREW_FILE))
  exit 0
end

require "pathname"
HOMEBREW_LIBRARY_PATH = Pathname.new(__FILE__).realpath.parent.join("Homebrew")
$:.unshift(HOMEBREW_LIBRARY_PATH.to_s)
require "global"

if ARGV.first == "--version"
  puts Homebrew.homebrew_version_string
  exit 0
elsif ARGV.first == "-v"
  puts "Homebrew #{Homebrew.homebrew_version_string}"
  # Shift the -v to the end of the parameter list
  ARGV << ARGV.shift
  # If no other arguments, just quit here.
  exit 0 if ARGV.length == 1
end

# Check for bad xcode-select before anything else, because `doctor` and
# many other things will hang
# Note that this bug was fixed in 10.9
if OS.mac? && MacOS.version < :mavericks && MacOS.active_developer_dir == "/"
  odie <<-EOS.undent
  Your xcode-select path is currently set to '/'.
  This causes the `xcrun` tool to hang, and can render Homebrew unusable.
  If you are using Xcode, you should:
    sudo xcode-select -switch /Applications/Xcode.app
  Otherwise, you should:
    sudo rm -rf /usr/share/xcode-select
  EOS
end

case HOMEBREW_PREFIX.to_s
when "/", "/usr"
  # it may work, but I only see pain this route and don't want to support it
  abort "Cowardly refusing to continue at this prefix: #{HOMEBREW_PREFIX}"
end

# Many Pathname operations use getwd when they shouldn't, and then throw
# odd exceptions. Reduce our support burden by showing a user-friendly error.
Dir.getwd rescue abort "The current working directory doesn't exist, cannot proceed."

def require?(path)
  require path
rescue LoadError => e
  # HACK: ( because we should raise on syntax errors but
  # not if the file doesn't exist. TODO make robust!
  raise unless e.to_s.include? path
end

begin
  trap("INT", std_trap) # restore default CTRL-C handler

  empty_argv = ARGV.empty?
  help_regex = /(-h$|--help$|--usage$|-\?$|^help$)/
  help_flag = false
  internal_cmd = true
  cmd = nil

  ARGV.dup.each_with_index do |arg, i|
    if help_flag && cmd
      break
    elsif arg =~ help_regex
      help_flag = true
    elsif !cmd
      cmd = ARGV.delete_at(i)
    end
  end

  cmd = HOMEBREW_INTERNAL_COMMAND_ALIASES.fetch(cmd, cmd)

  sudo_check = %w[ install link pin unpin upgrade ]

  if sudo_check.include? cmd
    if Process.uid.zero? and not File.stat(HOMEBREW_BREW_FILE).uid.zero?
      raise <<-EOS.undent
        Cowardly refusing to `sudo brew #{cmd}`
        You can use brew with sudo, but only if the brew executable is owned by root.
        However, this is both not recommended and completely unsupported so do so at
        your own risk.
        EOS
    end
  end

  # Add contributed commands to PATH before checking.
  Dir["#{HOMEBREW_LIBRARY}/Taps/*/*/cmd"].each do |tap_cmd_dir|
    ENV["PATH"] += "#{File::PATH_SEPARATOR}#{tap_cmd_dir}"
  end

  # Add SCM wrappers.
  ENV["PATH"] += "#{File::PATH_SEPARATOR}#{HOMEBREW_LIBRARY}/ENV/scm"

  if cmd
    internal_cmd = require? HOMEBREW_LIBRARY_PATH.join("cmd", cmd)

    if !internal_cmd && ARGV.homebrew_developer?
      internal_cmd = require? HOMEBREW_LIBRARY_PATH.join("dev-cmd", cmd)
    end
  end

  # Usage instructions should be displayed if and only if one of:
  # - a help flag is passed AND an internal command is matched
  # - a help flag is passed AND there is no command specified
  # - no arguments are passed
  #
  # It should never affect external commands so they can handle usage
  # arguments themselves.

  if empty_argv || (help_flag && (cmd.nil? || internal_cmd))
    # TODO: - `brew help cmd` should display subcommand help
    require "cmd/help"
    puts ARGV.usage
    exit ARGV.any? ? 0 : 1
  end

  if internal_cmd
    Homebrew.send cmd.to_s.gsub("-", "_").downcase
  elsif which "brew-#{cmd}"
    %w[CACHE CELLAR LIBRARY_PATH PREFIX REPOSITORY].each do |e|
      ENV["HOMEBREW_#{e}"] = Object.const_get("HOMEBREW_#{e}").to_s
    end
    exec "brew-#{cmd}", *ARGV
  elsif (path = which("brew-#{cmd}.rb")) && require?(path)
    exit Homebrew.failed? ? 1 : 0
  else
    onoe "Unknown command: #{cmd}"
    exit 1
  end

rescue FormulaUnspecifiedError
  abort "This command requires a formula argument"
rescue KegUnspecifiedError
  abort "This command requires a keg argument"
rescue UsageError
  onoe "Invalid usage"
  abort ARGV.usage
rescue SystemExit
  puts "Kernel.exit" if ARGV.verbose?
  raise
rescue Interrupt => e
  puts # seemingly a newline is typical
  exit 130
rescue BuildError => e
  e.dump
  exit 1
rescue RuntimeError, SystemCallError => e
  raise if e.message.empty?
  onoe e
  puts e.backtrace if ARGV.debug?
  exit 1
rescue Exception => e
  onoe e
  if internal_cmd
    puts "#{Tty.white}Please report this bug:"
    puts "    #{Tty.em}#{OS::ISSUES_URL}#{Tty.reset}"
  end
  puts e.backtrace
  exit 1
else
  exit 1 if Homebrew.failed?
end
